src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java

```diff
@@ -362,6 +374,69 @@ public String login(@ModelAttribute("authorizeCommand") @Validated AuthorizeRequ
+                                   String ipAddress,
+                                   HttpServletResponse response,
+                                   Model model) {
+        final String authIdCookie = spiService.initiateOtpeAuthentication(idamSessionCookie, ipAddress);
+        log.info("Successful OTP request - " + obfuscateEmailAddress(request.getUsername()));
+        response.addHeader(HttpHeaders.SET_COOKIE, makeCookieSecure(authIdCookie));
+
+        final VerificationRequest verificationRequest = new VerificationRequest();
+        verificationRequest.setUsername(request.getUsername());
+        verificationRequest.setResponse_type(request.getResponse_type());
+        verificationRequest.setState(request.getState());
+        verificationRequest.setClient_id(request.getClient_id());
+        verificationRequest.setRedirect_uri(request.getRedirect_uri());
+        verificationRequest.setScope(request.getScope());
+        verificationRequest.setSelfRegistrationEnabled(request.isSelfRegistrationEnabled());
+        model.addAttribute("verificationCommand", verificationRequest);
+
+        return VERIFICATION_VIEW;
+    }
+
+    @PostMapping("/verification")
+    public String verification(@ModelAttribute("verificationCommand") VerificationRequest request,
+                               BindingResult bindingResult,
+                               Model model,
+                               HttpServletRequest httpRequest,
+                               HttpServletResponse response) {
+        final String ipAddress = ObjectUtils.defaultIfNull(
+            httpRequest.getHeader(X_FORWARDED_FOR),
+            httpRequest.getRemoteAddr());
+
+        final String authIdCookieName = configurationProperties.getStrategic().getOtp().getOtpSessionIdCookie();
+        final String authIdCookie = Arrays.stream(ofNullable(httpRequest.getCookies()).orElse(new Cookie[] {}))
+            .filter(c -> authIdCookieName.equals(c.getName())) // find Idam.AuthId
+            .findAny() // get
+            .map(c -> String.format("%s=%s", c.getName(), c.getValue())) // map to: "Idam.AuthId=xyz"
+            .orElse(null); // get
+
+        model.addAttribute(RESPONSE_TYPE, request.getResponse_type());
+        model.addAttribute(STATE, request.getState());
+        model.addAttribute(CLIENT_ID, request.getClient_id());
+        model.addAttribute(REDIRECT_URI, request.getRedirect_uri());
+        model.addAttribute(SCOPE, request.getScope());
+        model.addAttribute(SELF_REGISTRATION_ENABLED, request.isSelfRegistrationEnabled());
+
+        final String cookie = spiService.submitOtpeAuthentication(authIdCookie, ipAddress, request.getCode());
+
+        final String responseUrl = authoriseUser(cookie, httpRequest);
+        final boolean loginSuccess = responseUrl != null && !responseUrl.contains("error");
+
+        if (loginSuccess) {
+            log.info("Successful login - " + obfuscateEmailAddress(request.getUsername()));
```

Comment:
might be worth having a different log message here indicating this is from the verification call